packages/flutter_tools/lib/src/device.dart

```diff
@@ -186,6 +189,30 @@ abstract class Device {
   String toString() => '$runtimeType $id';
 }
 
+class ForwardedPort {
+  ForwardedPort(this.hostPort, this.devicePort);
+
+  final int hostPort;
+  final int devicePort;
+
+  String toString() => 'ForwardedPort HOST:$hostPort to DEVICE:$devicePort';
+}
+
+/// Forward ports from the host machine to the device.
+abstract class DevicePortForwarder {
+  /// Returns a Future that completes with the current list of forwarded
+  /// ports for this device.
+  Future<List<ForwardedPort>> get forwardedPorts;
```

Comment:
I think this could `List<ForwardedPort>` w/o the Future - none of the impls need to be async.